src/transport_performance/gtfs/validation.py

```diff
@@ -601,3 +655,803 @@ def get_route_modes(self):
+            The name of the column contianing the
+            target data (counts)
+        orientation : str, optional
+            The orientation of the bar plot ("v" or "h"),
+            by default "v"
+        day_column : str, optional
+            The name of the column containing the day,
+            by default "day"
+        width : int, optional
+            The width of the plot (in pixels), by default 2000
+        height : int, optional
+            The height of the plot (in pixels), by default 800
+        xlabel : str, optional
+            The label for the x axis.
+            If left empty, the column name will be used,
+            by default None
+        ylabel : str, optional
+            The label for the y axis.
+            If left empty, the column name will be used,
+            by default None
+        plotly_kwargs : dict, optional
+            Kwargs to pass to fig.update_layout() for
+            additional plot customisation,
+            by default {}
+        return_html : bool, optional
+            Whether or not to return a html string,
+            by default False
+        save_html : bool, optional
+            Whether or not to save the plot as a html file,
+            by default False
+        save_image : bool, optional
+            Whether or not to save the plot as a PNG,
+            by default False
+        out_dir : Union[pathlib.Path, str], optional
+            The directory to save the plot into. If a file extension is added
+            to this directory, it won't be cleaned. Whatever is passed as the
+            out dir will be used as the parent directory of the save, leaving
+            the responsibility on the user to specify the correct path.,
+            by default os.path.join("outputs", "gtfs")
+        img_type : str, optional
+            The type of the image to be saved. E.g, .svg or .jpeg.,
+            by defauly "png"
+
+        Returns
+        -------
+        Union[PlotlyFigure, str]
+            Returns either a HTML string or the plotly figure
+
+        Raises
+        ------
+        ValueError
+            An error is raised if orientation is not 'v' or 'h'.
+        ValueError
+            An error is raised if an invalid iamge type is passed.
+
+        """
+        # parameter type defences
+        _type_defence(summary_df, "summary_df", pd.DataFrame)
+        _type_defence(day_column, "day_column", str)
+        _type_defence(target_column, "target_column", str)
+        _type_defence(plotly_kwargs, "plotly_kwargs", dict)
+        _type_defence(return_html, "return_html", bool)
+        _type_defence(width, "width", int)
+        _type_defence(height, "height", int)
+        _type_defence(xlabel, "xlabel", (str, type(None)))
+        _type_defence(ylabel, "ylabel", (str, type(None)))
+        _type_defence(save_html, "save_html", bool)
+        _type_defence(save_image, "save_iamge", bool)
+        _type_defence(img_type, "img_type", str)
```

Comment:
Consider whether a for loop will suit your needs in fewer lines:
```
for i in [day_column, target_column, xlabel, ylabel, img_type]:
  _type_defence(i, "grouped plot str inputs", (str, type(None)))
```

Reply:
**wont-fix**
Rationale: It takes approximately 3 lines (minimum) to check for 3 params using a loop which is the same as without a loop. This is because to maintain the correct error messages, a dictionary needs to be looped through.
There are no more than 3 type checks of one datatype, meaning using loops isn't beneficial.